plugins/async-node/ios/Sources/AsyncNodePlugin.swift

```diff
@@ -42,58 +43,58 @@ public class AsyncNodePlugin: JSBasePlugin, NativePlugin {
         self.plugins = plugins
     }
 
-    override public func setup(context: JSContext) {
-        super.setup(context: context)
+   override public func setup(context: JSContext) {
+           super.setup(context: context)
 
-        if let pluginRef = pluginRef {
-            self.hooks = AsyncNodeHook(onAsyncNode: AsyncHook(baseValue: pluginRef, name: "onAsyncNode"))
-        }
+           if let pluginRef = pluginRef {
+               self.hooks = AsyncNodeHook(onAsyncNode: AsyncHook2(baseValue: pluginRef, name: "onAsyncNode"))
+           }
 
-        hooks?.onAsyncNode.tap({ node in
-            // hook value is the original node
-            guard let asyncHookHandler = self.asyncHookHandler else {
-                return JSValue()
-            }
-
-            let replacementNode = try await (asyncHookHandler)(node)
-
-            switch replacementNode {
-            case .multiNode(let replacementNodes):
-                let jsValueArray = replacementNodes.compactMap({ node in
-                    switch node {
-                    case .concrete(let jsValue):
-                        return jsValue
-                    case .encodable(let encodable):
-                        let encoder = JSONEncoder()
-                        do {
-                            let res = try encoder.encode(encodable)
-                            return context.evaluateScript("(\(String(data: res, encoding: .utf8) ?? ""))") as JSValue
-                        } catch {
-                            return nil
-                        }
-                    }
-                })
-
-                return context.objectForKeyedSubscript("Array").objectForKeyedSubscript("from").call(withArguments: [jsValueArray])
-
-            case .singleNode(let replacementNode):
-                switch replacementNode {
-
-                case .encodable(let encodable):
-                    let encoder = JSONEncoder()
-                    do {
-                        let res = try encoder.encode(encodable)
-                        return context.evaluateScript("(\(String(data: res, encoding: .utf8) ?? ""))") as JSValue
-                    } catch {
-                        break
-                    }
-                case .concrete(let jsValue):
-                    return jsValue
-                }
-            }
-
-            return nil
-        })
+           hooks?.onAsyncNode.tap({ node, callback in
+               print("Value of callback \(callback)")
+               // hook value is the original node
+               guard let asyncHookHandler = self.asyncHookHandler else {
+                   return JSValue()
+               }
+
+               let replacementNode = try await (asyncHookHandler)(node)
+               var result: JSValue?
+
+               switch replacementNode {
```

Comment:
for lines 63 to 95 probably want to put this inside a function that takes a `AsyncNodeHandlerType` type so that the user can pass in those types from the `callback` in the test when we call `callback.call(withArguments: [` we can reuse this function